pkg/health/channel.go

```diff
@@ -0,0 +1,75 @@
+package health
+
+import "sync"
+
+type ChannelProvider struct {
+	expected     map[Check]struct{}
+	tr           Tracker
+	OnReady      chan struct{}
+	OnTerminated chan struct{}
+
+	sendReady      func()
+	sendTerminated func()
+}
+
+var _ Provider = (*ChannelProvider)(nil)
+
+func NewChannelProvider(
+	tr Tracker,
+	opts ...CheckOption,
+) *ChannelProvider {
+	o := CheckOptions{}
+	o.Apply(opts...)
+
+	c := &ChannelProvider{
+		tr:           tr,
+		expected:     o.expected,
+		OnReady:      make(chan struct{}, 2),
```

Comment:
could these just be closed instead of sending a value?

Reply:
WDYT?